core/tee/tee_svc_storage.c

```diff
@@ -598,7 +639,42 @@ TEE_Result tee_svc_storage_obj_create(uint32_t storage_id, void *object_id,
 		goto exit;
 	}
 
+	file = tee_svc_storage_create_filename(sess, object_id,
+					       object_id_len, false);
+	if (file == NULL) {
+		res = TEE_ERROR_OUT_OF_MEMORY;
+		goto exit;
+	}
+
+	err = tee_file_ops.access(file, TEE_FS_F_OK);
+	if (!err) {
+		/* file exists */
+		if (flags & TEE_DATA_FLAG_OVERWRITE) {
+			goto rename;
+		} else {
+			res = TEE_ERROR_ACCESS_CONFLICT;
+			goto clean;
+		}
+	}
+
+rename:
+	err = -1;
+	err = tee_file_ops.rename(initfile, file);
+	if (err) {
+		/* error codes needs better granularity */
+		res = TEE_ERROR_GENERIC;
+		goto clean;
+	}
+
+clean:
+	tee_file_ops.unlink(initfile);
+
 exit:
+	free(file);
+	file = NULL;
+	free(initfile);
+	initfile = NULL;
+
```

Comment:
In the following line, we do `tee_file_ops.close(fd);`. Should not it be performed before the rename?